public/js/app/components/SyllabusInformation.jsx

```diff
@@ -1,39 +1,35 @@
 import React from 'react'
 import { Col } from 'reactstrap'
 import { FaAsterisk } from 'react-icons/fa'
+import { useLanguage } from '../hooks/useLanguage'
 
-const SyllabusInformation = ({ context = {}, syllabus = {}, hasSyllabus, translation = {} }) => {
-  const { courseCode } = context
+const ActualSyllabusInformation = ({ syllabusName }) => {
+  const { translation } = useLanguage()
 
-  const { courseLabels, courseInformation } = translation
   const {
     label_course_syllabus_source: labelCourseSyllabusSource,
     label_course_syllabus_denoted: labelCourseSyllabusDenoted,
     syllabus_marker_aria_label: labelCourseSyllabusAsterisk,
-  } = courseLabels
-  const { course_short_semester: courseShortSemester } = courseInformation
-
-  const { course_valid_from: courseValidFrom = ['', ''], course_valid_to: courseValidTo = ['', ''] } = syllabus
-  const courseValidFromLabel = `${courseShortSemester[courseValidFrom[1]]}${courseValidFrom[0]}`
-  const courseValidToLabel = courseValidTo.length ? courseShortSemester[courseValidTo[1]] + '' + courseValidTo[0] : ''
-  const courseValidRangeLabel = `(${courseValidFromLabel}\u2013${courseValidToLabel})`
-
-  const syllabusText = `${labelCourseSyllabusSource} ${courseCode} ${courseValidRangeLabel} ${labelCourseSyllabusDenoted}`
+  } = translation.courseLabels
 
   return (
     <Col sm="12">
       <div>
-        {hasSyllabus && (
-          <span>
-            {syllabusText}
-            {' ( '}
-            <FaAsterisk className="syllabus-marker-icon-small" aria-label={labelCourseSyllabusAsterisk} />
-            {' )'}
-          </span>
-        )}
+        <span>
+          {`${labelCourseSyllabusSource} ${syllabusName} ${labelCourseSyllabusDenoted}`}
+          {' ( '}
+          <FaAsterisk className="syllabus-marker-icon-small" aria-label={labelCourseSyllabusAsterisk} />
+          {' )'}
+        </span>
       </div>
     </Col>
   )
 }
 
-export default SyllabusInformation
+const SyllabusInformationOrNull = ({ syllabusName }) => {
+  if (!syllabusName) return null
+
+  return <ActualSyllabusInformation syllabusName={syllabusName} />
+}
+
+export { SyllabusInformationOrNull as SyllabusInformation }
```

Comment:
Curious what you think about this solution
I thought that calling the component `SyllabusInformationOrNull` for `SyllabusInformation` would be misleading. Rather the component `ActualSyllabusInformation` should be called `SyllabusInformation`, but then we would have a mismatch in file name <-> exported component. 

Note also the change to named export.

Reply:
First: I like the change to named export. What are the real advantages of using default export? Or why don't we always use named exports?

I'm not sure I think calling `SyllabusInformationOrNull` for `SyllabusInformation` would be misleading actually. For the "user"  `SyllabusInformation` is what the component actually is. The null check with an extra component is just an internal behaviour the "user" doesn't need to know about ("user"=the other component using this component). 

And with that said, the only like important thing is the name of the actual exported component.

Reply:
Just curious about the approach here.

Why can’t the functional component ActualSyllabusInformation just be called SyllabusInformation and either just return the html if syllabusName exists or  return (null) if syllabusName does not exist? Is it not possible to get rid of SyllabusInformationOrNull altogether?

Reply:
Very good question!
This is certainly possible. This is a rather philosophical discussion about readability and the notion that "one function should do one thing", which - I think - is mentioned in `Clean Code` in the chapter about functions.

Reply:
I completely agree with Karl and think the approach that you have used is indeed the cleanest approach that could be used here.
right now there is nothing that mislead other developers. I also really liked the aliasing approach for exporting the component.
I used to handle this functionality (checking to show the component if is not null) inline in the other components and now I think this approach can help me to have cleaner code. 
I'll also check the mentioned part inside the clean code book which is valuable to read.